packages/Syntax/src/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Takes an object type or interface as a parameter, and
+ * returns a union type composed of the values of each property.
+ *
+ * ```
+ * const MyObject = { First: 'first', Second: 'second' }
+ * type MyObjectType = StringUnionFromObjectType<typeof MyObject>
+ *
+ * // MyObjectType: 'first' | 'second'
+ * ```
+ */
+type UnionFromValues<T> = T[keyof T];
```

Comment:
Yeah... I'd thought a little about this too but I went this route erring on the side of DRYness. Perhaps I was too close to this code for too long and found it more readable because of that. I'll switch it out – I don't think the slightly cleaner syntax outweighs the code readability hit looking back at this.